app/models/user.rb

```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  validates :name, presence: true
+  validates :posts_count, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```

Comment:
- [ ] You are displaying the users' posts with the most recent comments on the `post` index page. Looks like you are making two different queries for this and this is causing an N+1 problem.  You can include comments to your post query to solve this N+1 issue. 

Please check the below screenshots

![Screenshot from 2021-12-03 20-55-49](https://user-images.githubusercontent.com/1642014/144650854-15a2c96e-24c1-45e1-ac38-dfc15a9d45e8.png)
Currently, the application makes 4 SQL queries, 1 for posts and 3 for comments of each post. This significantly reduces your application performance (Total query time 12.3 ms)

![Screenshot from 2021-12-03 20-59-45](https://user-images.githubusercontent.com/1642014/144650862-e92ecec7-e984-4e2b-ac66-ce94ba86a25b.png)
If you solve the N+1 issue your query will look like this, just two queries for posts and comments. (Total query time 4.2ms it is **3** times faster than the query with N+1 issue)

Reply:
Addressed this!